analyzers/src/SonarAnalyzer.Common/SymbolicExecution/Roslyn/RoslynSymbolicExecution.cs

```diff
@@ -89,11 +93,14 @@ private IEnumerable<ExplodedNode> ProcessBranching(ExplodedNode node)
             }
             else if (node.Block.ContainsThrow())
             {
+                node = new ExplodedNode(cfg.ExitBlock, CleanUnusedState(node.State, node.Block), null);
                 yield return CreateNode(cfg.ExitBlock, null, null);
             }
             else
             {
-                foreach (var successor in node.Block.Successors.Where(x => IsReachable(node, x)))
+                var reachableSuccessors = node.Block.Successors.Where(x => IsReachable(node, x)).ToList();
+                node = new ExplodedNode(node.Block, CleanUnusedState(node.State, node.Block), node.FinallyPoint);
+                foreach (var successor in reachableSuccessors)
```

Comment:
We should discuss it offline, it seems to me that the current implementation is indeed the place where a block is finished. The ProcessBranch method is the "entrance" to the block, so it can be called there but that means we will change a bit the logic, we will clean the symbols that are not "LiveIn" for the block.

Reply:
Clarified offline.